src/RouteCollector.php

```diff
@@ -68,17 +89,30 @@ public function route(
         array $methods = null,
         string $name = null
     ) : Route {
-        $this->checkForDuplicateRoute($path, $methods);
 
-        $methods = null === $methods ? Route::HTTP_METHOD_ANY : $methods;
+        $methods = $methods ?? Route::HTTP_METHOD_ANY;
         $route   = new Route($path, $middleware, $methods, $name);
-
+        $this->checkForDuplicateRoute($route);
+        $this->fillRouteSearchStructure($route);
         $this->routes[] = $route;
         $this->router->addRoute($route);
 
         return $route;
     }
 
+    private function fillRouteSearchStructure(Route $route): void
+    {
+        $this->routeNames[$route->getName()] = $route;
+        if ($route->allowsAnyMethod()) {
+            $this->routePaths[$route->getPath()]['any'] = $route;
```

Comment:
Moved to constant. I dont like it either, but theoretically we can have custom HTTP method `ANY` and it can have collision with this key if we put it inside other methods search structure. Do you have any other ideas?